projectLive/R/mod_summary_snapshot.R

```diff
@@ -155,65 +155,50 @@ mod_summary_snapshot_server <- function(
     create_info_box(param_list, group_object())
   })
   
-  output$study_per_consortium <- plotly::renderPlotly({
+  output$consortium_activity <- plotly::renderPlotly({
     shiny::req(data_config, group_object())
     param_list <- purrr::pluck(
       data_config,
       "modules",
       "summary_snapshot",
       "outputs",
-      "study_per_consortium"
+      "consortium_activity"
     )
     
-    data <- group_object()[[param_list$table]] %>% 
-      concatenate_df_list_columns_with_param_list(param_list) %>%    
-      recode_df_with_param_list(param_list) %>% 
-      rename_df_columns_with_param_list(param_list)
+    data <- group_object() %>% 
+      purrr::pluck(param_list$table) %>% 
+      format_plot_data_with_param_list(param_list)
     
-    validate(need(
-      nrow(data) > 0 , 
-      "The investigator/investigators has/have not uploaded any files yet. Please check back later."
-    ))
+    shiny::validate(shiny::need(nrow(data) > 0, param_list$empty_table_message))
     
-    create_study_per_consortium_plot(
-      data  = data, 
-      x     = param_list$columns$x$display_name,
-      fill  = param_list$columns$fill$display_name,
-      param_list$columns$facet$display_name
-    ) %>% 
-      plotly::ggplotly(
-        tooltip = c("count", "fill")
-      )
+    create_plot_with_param_list(
+      data,
+      param_list,
+      "create_consortium_activity_plot"
+    )
   })
   
   
-  output$files_per_study <- plotly::renderPlotly({
+  output$resources_generated <- plotly::renderPlotly({
     shiny::req(data_config, group_object())
     param_list <- purrr::pluck(
       data_config,
       "modules",
       "summary_snapshot",
       "outputs",
-      "files_per_study"
+      "resources_generated"
     )
     
-    data <- group_object()[[param_list$table]] %>% 
-      concatenate_df_list_columns_with_param_list(param_list) %>%    
-      recode_df_with_param_list(param_list) %>% 
-      rename_df_columns_with_param_list(param_list)
+    data <- group_object() %>% 
+      purrr::pluck(param_list$table) %>% 
+      format_plot_data_with_param_list(param_list)
+      
+    shiny::validate(shiny::need(nrow(data) > 0, param_list$empty_table_message))
     
-    validate(need(
-      nrow(data) > 0, 
-      "The investigator/investigators has/have not uploaded any files yet. Please check back later."
-    ))
-    create_files_per_study_plot(
-      data  = data,
-      x     = param_list$columns$x$display_name,
-      fill  = param_list$columns$fill$display_name,
-      param_list$columns$facet$display_name
-    ) %>%
-      plotly::ggplotly(
-        tooltip = c("count", "fill")
-      )
+    create_plot_with_param_list(
```

Comment:
For this plot, we can have the legends below the plot with a legend title informing users that if they click on the legends they can see how the study generated resources over time.

Reply:
I added a layout call that moves the legend to the bottom. I don't think it looks great this way however.
 I couldn't figure out how to change the legend title.

Reply:
I think this would be helpful in changing the legend title: 
> fig <- fig %>% layout(legend=list(title=list(text='<b> Trend </b>')))

Reply:
For this one: X axis label should be "Number of files per resource"

Reply:
That did work, but I don't think it looks good. I'll let you decide however :)

Reply:
Yeah, I agree with you ... the legends below doesn't look so great :(

Reply:
Lets keep the legends on the side for now...

Reply:
For this one: X axis label should be "Number of files"

Reply:
The legend is quite distracting for the plot. Lets hide the legend for this plot since you can see the name of the study in the hover. 

We can add it back once we resolve the legend position issue.